### PR TITLE
Gmail Rest Connectcor

### PR DESCRIPTION
**Pull request by ashalya**

### CODE REVIEW

File: gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/GmailSMTPConnectionObject.java [hunk 1]
```diff
@@ -1,66 +0,0 @@
-/*
- * Copyright (c) 2005-2013, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (kesavany) [inline gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/GmailSMTPConnectionObject.java]: Change the license header.

File: gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/test/resources/artifacts/ESB/config/proxies/gmail/gmail_listDrafts.xml [hunk 2]
```diff
@@ -1,50 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-        <!--
```
Comment (kesavany) [inline gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/test/resources/artifacts/ESB/config/proxies/gmail/gmail_listDrafts.xml]: Format the code.

File: gmailRest/gmailRest-connector/gmailRest-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/gmailRest.java [hunk 3]
```diff
@@ -20,12 +20,12 @@
 import org.apache.synapse.MessageContext;
 import org.wso2.carbon.connector.core.*;
 
-public class gmail extends AbstractConnector {
+public class gmailRest extends AbstractConnector {
```
Comment (Shakila) [inline gmailRest/gmailRest-connector/gmailRest-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/gmailRest.java]: This class is not used. Can we remove this?

File: gmailRest/gmailRest-connector/gmailRest-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/config/gmailRestinit.xml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  ~  Unless required by applicable law or agreed to in writing,
+  ~  software distributed under the License is distributed on an
+  ~  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~  KIND, either express or implied.  See the License for the
+  ~  specific language governing permissions and limitations
+  ~  under the License.
+  -->
+
+<template name="gmailRestinit" xmlns="http://ws.apache.org/ns/synapse">
+    <parameter name="access_token" description="URL of the nimble API"/>
```
Comment (Shakila) [inline gmailRest/gmailRest-connector/gmailRest-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/config/gmailRestinit.xml]: Description is wrong and parameter name should be in camel-case format.

File: gmailRest/gmailRest-connector/gmailRest-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/gmailRest.java (hunk 3 above)
Comment (ashalya) [inline gmailRest/gmailRest-connector/gmailRest-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/gmailRest.java]: Fixed

File: gmailRest/gmailRest-connector/gmailRest-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/config/gmailRestinit.xml (hunk 4 above)
Comment (ashalya) [inline gmailRest/gmailRest-connector/gmailRest-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/config/gmailRestinit.xml]: Fixed

File: gmailRest/gmailRest-connector/gmailRest-connector-1.0.0/org.wso2.carbon.connector/src/test/java/org/wso2/carbon/connector/integration/test/gmailRest/GmailConnectorIntegrationTest.java [hunk 5]
```diff
@@ -1,13 +1,8 @@
-package org.wso2.carbon.connector.integration.test.gmail;
```
Comment (ashalya) [inline gmailRest/gmailRest-connector/gmailRest-connector-1.0.0/org.wso2.carbon.connector/src/test/java/org/wso2/carbon/connector/integration/test/gmailRest/GmailConnectorIntegrationTest.java]: licence header added in https://github.com/wso2/esb-connectors/pull/532

File: gmailRest/gmailRest-connector/gmailRest-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/GmailSMTPConnectionObject.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+package org.wso2.carbon.connector;
+
+import com.google.code.com.sun.mail.smtp.SMTPTransport;
+import com.google.code.javax.mail.Session;
+
+/**
+ * This class provides the object model to store {@link com.google.code.javax.mail.Session} and
+ * {@link com.google.code.com.sun.mail.smtp.SMTPTransport} instances to send mails through SMTP.
+ */
+public class GmailSMTPConnectionObject {
```
Comment (prabathabey) [inline gmailRest/gmailRest-connector/gmailRest-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/GmailSMTPConnectionObject.java]: Classes are considered "blue prints" of objects. So, naming a class with the notion of an "object" is inappropriate.

**Pull request merged**